dashboard/src/modules/components/TableComponent/index.jsx

```diff
@@ -134,113 +135,119 @@ const TableWithFavorite = () => {
-        <PathBreadCrumb pathList={datasetBreadcrumb} />
-        <Heading
-          containerClass="publicDataPageTitle"
-          headingTitle="Results"
-        ></Heading>
-        <div className="filterContainer">
-          <SearchBox
-            dataArray={tableData}
-            setPublicData={setPublicData}
-            startDate={startDate}
-            endDate={endDate}
-            setDatasetName={setDatasetName}
-            aria-label="search box"
-          />
-          <DatePickerWidget
-            dataArray={tableData}
-            setPublicData={setPublicData}
-            datasetName={datasetName}
-            setDateRange={setDateRange}
-            aria-label="date picker"
-          />
-        </div>
-        <ToggleGroup aria-label="Result Selection Options">
-          <ToggleGroupItem
-            text={`All Results(${publicData?.length})`}
-            buttonId="datasetListButton"
-            isSelected={isSelected === "datasetListButton"}
-            onChange={handleButtonClick}
-            className="datasetListButton"
-            aria-label="see dataset button"
-          />
-          <ToggleGroupItem
-            text={`Favorites(${favoriteRepoNames?.length})`}
-            buttonId="favoriteListButton"
-            isSelected={isSelected === "favoriteListButton"}
-            onChange={handleButtonClick}
-            className="favoriteListButton"
-            aria-label="see favorites button"
-          />
-        </ToggleGroup>
-        <div className="table-scroll-container">
-          <OuterScrollContainer>
-            <InnerScrollContainer>
-              <TableComposable
-                aria-label="Favoritable table"
-                variant="compact"
-                isStickyHeader
-              >
-                <Thead>
-                  <Tr>
-                    <Th sort={getSortParams(1)}>{columnNames.name}</Th>
-                    <Th sort={getSortParams(2)}>{columnNames.creationDate}</Th>
-                    <Th sort={getSortParams(3)}></Th>
-                  </Tr>
-                </Thead>
-                <Tbody>
-                  {selectedArray.length > 0 ? (
-                    selectedArray.map((repo, rowIndex) => (
-                      <Tr key={rowIndex}>
-                        <Td dataLabel={columnNames.name}>{repo.name}</Td>
-                        <Td dataLabel={columnNames.creationDate}>
-                          {repo.metadata["dataset.created"]}
-                        </Td>
-                        <Td
-                          favorites={{
-                            isFavorited: isRepoFavorited(repo),
-                            onFavorite: (_event, isFavoriting) => {
-                              markRepoFavorited(repo, isFavoriting);
-                            },
-                            rowIndex,
-                          }}
-                        />
+      {!loginDetails?.isLoggedIn && (
```

Comment:
If the user is logged in, Overview page should be rendered which is under development.

Reply:
Shouldn't that be handled by conditional *routing* to a different component rather than conditionalizing the `TableComponent`?

In general, we want *page views* to work regardless of whether you're logged in; it's just that if you're not, you can only see "public" datasets, and can't change anything. The "Overview" page (and as you implemented elsewhere the Profile view) is an exception as they have no meaning if you're not logged in. (Although a user who hasn't logged in should probably be able to select the overview page and get side-tracked to login before showing anything.)

In particular, a user who logs in should still be able to browse other accessible data, not just "curate" their own data, and there's nothing about this page that shouldn't work in that environment, right?

Reply:
Let's have the separate route for the "public" datasets which the user can view irrespective of being logged in. 
Let's take that separately. 

Tracked in [PBENCH-757](https://issues.redhat.com/browse/PBENCH-757)

Reply:
I suspect that this added conditional is not accomplishing anything.  (That is, if the user is not logged in, then I think we get redirected at the app level and never end up doing much here.)  So, I'm not sure that it's worth adding the extra complication here.

Nevertheless, this function is really, really hard to read.  I'm not saying that we should fix it in this PR, but this function should be broken up with each of the conditional pieces relegated to a subroutine or other external definition, so that the structure is much more clear -- "if this condition is true, then we load that; otherwise, we load this other" -- having the tags in-line hides the structure.  (Ternaries are good when it's easy to see the alternate values, e.g., when they fit on one line or so; at this scale, they are evil. 😈)

Reply:
I'm not entirely comfortable with this, but I'm willing to defer it for another PR. @webbnh are you ready for me to resolve this, or is this something you'd like to push further? 😄